// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";


package zen_notice;

//  
service ZenNoticeHandle {
  rpc NoticeLine(NoticeLineReq) returns (NoticeLineRsp) {}
  rpc NoticeDraw(NoticeDrawReq) returns (NoticeDrawRsp) {}
  rpc NoticeTrade(NoticeTradeReq) returns (NoticeTradeRsp) {}
}

//  
message NoticeLineReq {
  string code = 1;
  string name = 2;
  string period = 3;
  int32 trigger_ts = 4;
  int32 direction = 5;
  int32 line_start_ts = 6;
  int32 line_end_ts = 7;
  double high = 8;
  double low = 9;
  string sec_type = 10;
  int32 kline_start_ts = 11;
}

message NoticeLineRsp {
  int32 status = 1;
  string message = 2;
}

message NoticeDrawReq {
  string code = 1;
  string name = 2;
  string period = 3;
  int32 kline_start_ts = 4;
}

message NoticeDrawRsp {
  int32 status = 1;
  string message = 2;
}

message NoticeTradeReq {
  string code = 1;
  string sec_type = 2;
  string name = 3;
  string period = 4;
  int32 trade_type = 5;
  int32 open_cover_type = 6;
  int32 direction = 7;
  int32 trigger_ts = 8;
  int32 kline_start_ts = 9;
  double price = 10;
  double stop_loss_price = 11;
  string ext_info = 12;
}

message NoticeTradeRsp {
  int32 status = 1;
  string message = 2;
}
